File: abc/abc352/a.java
// https://atcoder.jp/contests/abc352/submissions/53138403

import java.util.*;

class Main {
  public static void main(String args[]) {
    Scanner stdIn = new Scanner(System.in);
    
    int n = stdIn.nextInt();
    int x = stdIn.nextInt();
    int y = stdIn.nextInt();
    int z = stdIn.nextInt();

    if(x > y) {
      int t = x;
      x = y;
      y = t;
    }

    System.out.print(x <= z && z < y ? "Yes" : "No");
  }
}

File: abc/abc352/a.cpp
// https://atcoder.jp/contests/abc352/submissions/53167200

#include <bits/stdc++.h>
using namespace std;

int main() {
  int n, x, y, z;
  cin >> n >> x >> y >> z;

  if(x > y) {
    swap(x, y);
  }

  cout << (x < z && z < y ? "Yes" : "No");
}
